### ISSUE TITLE
Cabal 3.12.1.0 install failure for `lib:Agda` - dist/build/agda/agda does not exist

### ISSUE THREAD

**Issue by cho-m** [agda/agda#7401]
When trying to upgrade `cabal-install` in Homebrew to latest 3.12.1.0 (https://github.com/Homebrew/homebrew-core/pull/178592), I saw a Cabal install failure for `lib:Agda` (which was being installed as dependency for agda2hs).

https://github.com/Homebrew/homebrew-core/actions/runs/10114758622/job/27984382041#step:3:293
```
  [429 of 429] Compiling Agda.Main        ( src/full/Agda/Main.hs, dist/build/Agda/Main.o, dist/build/Agda/Main.dyn_o )
  Installing library in /opt/homebrew/Cellar/agda/2.6.4.3_1/libexec/ghc-9.8.2-inplace/incoming/new-29976/opt/homebrew/Cellar/agda/2.6.4.3_1/libexec/ghc-9.8.2-inplace/Agd-2.6.4.3-5231c10e/lib
  Generating Agda library interface files...
  dist/build/agda/agda: readCreateProcess: posix_spawnp: does not exist (No such file or directory)
  Error: [Cabal-7125]
  Failed to build Agda-2.6.4.3 (which is required by exe:agda2hs from agda2hs-1.2). See the build log above for details.
```

---

The corresponding versions in CI were for Agda 2.6.4.3 building with Homebrew's GHC 9.8.2 on arm64 macOS Ventura. The exact same setup with Cabal 3.10.3.0 had no issue.

I was also able to reproduce issue by just running `cabal install --lib` on master branch of Agda (currently https://github.com/agda/agda/commit/5da3f3f2be5fc0d4ed6a2b828bd751e44e29292f), e.g.
```console
❯ cabal --version
cabal-install version 3.12.1.0
compiled using version 3.12.1.0 of the Cabal library

❯ cabal update
Downloading the latest package list from hackage.haskell.org
Package list of hackage.haskell.org has been updated.
The index-state is set to 2024-07-27T12:29:48Z.

❯ cabal install --lib
...
 - Agda-2.8.0 (lib:Agda) (requires build)
...
Starting     Agda-2.8.0 (all, legacy fallback: build-type is Custom)
Building     Agda-2.8.0 (all, legacy fallback: build-type is Custom)
Installing   Agda-2.8.0 (all, legacy fallback: build-type is Custom)

Failed to build Agda-2.8.0. The failure occurred during the final install
step.
...
Generating Agda library interface files...
dist/build/agda/agda: readCreateProcess: posix_spawnp: does not exist (No such file or directory)
Error: [Cabal-7125]
Failed to build Agda-2.8.0. See the build log above for details.
```

If only building but not installing lib (e.g. `cabal build` or default `exe:agda, exe:agda-mode`), there is no issue.

**Comment by andreasabel** [comment]
In general `cabal install --lib` is not well supported by `cabal`, and it seems it does not work with Agda's custom `Setup.hs` script which wants to run Agda to compile the core library during installation.

@cho-m If you want a global install of Agda-the-library you can use `cabal v1-install`.

**Comment by cho-m** [comment]
`cabal install --lib` was mainly to easily reproduce isolated to Agda repository.

The original failure was installing executable for Agda2hs (https://github.com/agda/agda2hs).

**Comment by andreasabel** [comment]
What we are seeing here is actually a regression in `cabal-3.12.1.0`.  It still worked with `cabal-3.10.3.0`.

Upstream issue:
- https://github.com/haskell/cabal/issues/10235

**Comment by andreasabel** [comment]
I wonder whether we should work around this cabal issue in Agda 2.7.0.
We could for instance just emit a warning if the Agda executable is not available during installation, rather than failing the whole installation.   This would help with `cabal install Agda --lib`, but make install failures less discoverable in situation where the interface files _must_ be built, e.g. for system-wide installs into `root` space.

**Comment by andreasabel** [comment]
> We could for instance just emit a warning if the Agda executable is not available during installation, rather than failing the whole installation.

I implemented this in #7404.
